### PR TITLE
Added some logging to delete_topic_i so that when the topic is not re…

### PR DESCRIPTION
**Pull request by jwillemsen**
…moved from the map we can see that in the log, related to issue #2062

    * dds/DCPS/DomainParticipantImpl.cpp:

### CODE REVIEW

**Comment by jwillemsen** [comment]
@mitza-oci Maybe this one also for 3.15?

File: dds/DCPS/DomainParticipantImpl.cpp [hunk 1]
```diff
@@ -620,10 +618,9 @@ DomainParticipantImpl::delete_topic_i(
         return DDS::RETCODE_ERROR;
       }
 
-      --entry->client_refs_;
+      const CORBA::ULong client_refs = --entry->client_refs_;
 
-      if (remove_objref == true ||
-          0 == entry->client_refs_) {
+      if (remove_objref == true || 0 == client_refs) {
```
Comment (mitza-oci) [inline dds/DCPS/DomainParticipantImpl.cpp:623]:
```suggestion
      if (remove_objref || 0 == client_refs) {
```

File: dds/DCPS/DomainParticipantImpl.cpp [hunk 2]
```diff
@@ -650,9 +647,16 @@ DomainParticipantImpl::delete_topic_i(
           }
           return DDS::RETCODE_ERROR;
 
-        } else
+        } else {
           return DDS::RETCODE_OK;
-
+        }
+      } else {
+        if (DCPS_debug_level >= 1) {
```
Comment (mitza-oci) [inline dds/DCPS/DomainParticipantImpl.cpp:654]: Should this be logged at a higher level?  I guess I'm not familiar with the issue to understand why someone would be interested in this log message.

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: dds/DCPS/DomainParticipantImpl.cpp (hunk 2 above)
Comment (jwillemsen) [reply dds/DCPS/DomainParticipantImpl.cpp:654]: We could do that, which level, 4?

Comment (iguessthislldo) [reply dds/DCPS/DomainParticipantImpl.cpp:654]: I've never been quite sure what [it means by "administrative interfaces"](https://github.com/objectcomputing/OpenDDS/blob/c3e21a83403b91ea06e532506932bc9f8339722e/dds/DCPS/debug.h#L28).

Comment (mitza-oci) [reply dds/DCPS/DomainParticipantImpl.cpp:654]: My guess would be things that are in between Entity creation/deletion and per-sample.  So other Entity "controls" like set_qos?

**Review by iguessthislldo — APPROVED** [review]

**Review by mitza-oci — APPROVED** [review]

**Comment by mitza-oci** [comment]
Merge conflict from #2181

**Pull request merged**